verifiers/interfaceValidator.js

```diff
@@ -44,17 +44,52 @@ function getRamlFilePath(filePath) {
 }
 
 // Get all of the methods defined in a raml schema file.
-// TODO: update how the names are assembled to handle multiple parts and to ignore
-// parameter values.
 function getSchemaMethods(ramlPath) {
+
+  function addMethod(methodName) {
+    if(ramlMethods.indexOf(methodName) === -1) {
+      ramlMethods.push(methodName);
+    }
+  }
+
   var ramlMethods = [];
 
   if(ramlPath) {
-    raml.loadApiSync(ramlPath).resources().forEach(resource => {
-      resource.methods().forEach(method => {
-        ramlMethods.push(method.method() + resource.completeRelativeUri().substring(1));
-      });
+    var resources = [];
+    raml.loadApiSync(ramlPath).resources().forEach(r => {
+      resources.push(r);
     });
+
+    while(resources.length > 0) {
+      var current = resources.shift();
+      var resourceParts = current.completeRelativeUri().substring(1).split("/");
+      var suffix = "";
+
+      resourceParts.forEach(part => {
+        if(!part.startsWith("{") && !part.startsWith("?")){
+          suffix += part.charAt(0).toUpperCase() + part.slice(1);
+        }
+      });
+
+      current.is().forEach(trait => {
+        var traitName = trait.name();
+        if(traitName === "interface-sensor") {
+          addMethod("get" + suffix);
+        }
+        else if(traitName === "interface-actuator") {
+          addMethod("get" + suffix);
+          addMethod("post" + suffix);
+        }
+      });
+
+      current.methods().forEach(method => {
+        addMethod(method.method() + suffix);
+      });
+
+      current.resources().forEach(child => {
+        resources.push(child);
+      });
+    }
   }
 
```

Comment:
How do we validate that any GET or POST calls in the thingTranslator.js are also defined in the RAML? For instance, refresh mechanism was added to Wink but it got left out from RAML. We will fix this for now but it would be good to catch any such future instances.